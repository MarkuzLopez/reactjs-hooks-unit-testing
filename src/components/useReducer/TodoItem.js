import React from 'react';

const TodoItem = (props) => {

    const { todo, index, toggleTodo, deleteTodo } = props;

    return (
        <li key={todo.id} className="list-group-item flex-box" >
        <p className={`${todo.done && 'complete'} pointer`} onClick={ () => toggleTodo(todo.id) } >{index + 1}. {todo.desc}</p>
        <button 
        className="btn btn-danger"
        type="button"
        onClick={ () => deleteTodo(todo.id)}
        >Borrar</button>
        </li>
     );
}
 
export default TodoItem;