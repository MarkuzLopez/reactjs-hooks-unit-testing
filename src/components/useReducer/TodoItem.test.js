import { shallow } from "enzyme";
import { demoTodos } from '../../tests/fixtures/demosTodos';
import TodoItem from "./TodoItem";

describe('pruebas del componente todoList', () => {
    const  toggleTodo = jest.fn();
    const deleteTodo = jest.fn();
    const wrapper = shallow(
        <TodoItem
            todo={demoTodos[0]}
            index={0}
            toggleTodo={toggleTodo}
            deleteTodo={deleteTodo}
        />
    )


    test('debe de crear snapshots y mostrarse correctamente', () => {
       expect(wrapper).toMatchSnapshot();
    });

    test('debe de llamar la funcion de deleteToggle ', () => {

        wrapper.find('button').simulate('click');
        expect(deleteTodo).toHaveBeenCalledWith(demoTodos[0].id);
    });
    
    test('debe de llamar la funcion de toggleTodo', () => {
        
        wrapper.find('p').simulate('click');
        expect(toggleTodo).toHaveBeenCalledWith(demoTodos[0].id);
    });
    
    test('debe de mostrar el texto correctamente ', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(`1. ${demoTodos[0].desc}`);
    });
    
    test('debe de tener la clase complete si el TODO.done =  true', () => {
        const todo = demoTodos[0];
        todo.done =  true;

        const wrapper =  shallow(
            <TodoItem 
                todo={todo}
            />
        );

        // console.log(wrapper.html());
        expect(wrapper.find('p').hasClass('complete')).toBe(true);
    })
    
})
