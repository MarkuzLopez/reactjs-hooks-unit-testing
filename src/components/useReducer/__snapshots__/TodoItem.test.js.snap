// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas del componente todoList debe de crear snapshots y mostrarse correctamente 1`] = `
<li
  className="list-group-item flex-box"
  key="1"
>
  <p
    className="false pointer"
    onClick={[Function]}
  >
    1
    . 
    Aprender React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    Borrar
  </button>
</li>
`;
